src/main/java/com/monitorjbl/xlsx/StreamingReader.java

```diff
@@ -111,11 +127,57 @@ private void handleEvent(XMLEvent event) throws SAXException {
         currentCell = new StreamingCell(CellReference.convertColStringToIndex(coord[0]), Integer.parseInt(coord[1]) - 1);
 
         Attribute type = startElement.getAttributeByName(new QName("t"));
+/*ForgetMeNot--ibell--20150911--old code which didn't determine numeric vs date data types.
         if(type != null) {
           currentCell.setType(type.getValue());
         }
+*/
+        //
+        // This block will determine the value type and set the appropriate
+        // XSSF data type (used to format numeric cell values).
+        //
+        final String typeValue = (type != null) ? type.getValue() : "";
```

Comment:
again, copied and adapted from the referenced source.

I only offered this as a fix to the [perceived] shortcoming in your original solution (i.e., not considering styles/formatting of raw cell values).  It won't hurt my feelings if you dismiss it.